gdpr/impl.go

```diff
@@ -142,6 +143,12 @@ func (p *permissionsImpl) allowPITCF2(parsedConsent api.VendorConsents, vendor a
 	} else {
 		allowGeo = true
 	}
+	for i := 2; i <= 10; i++ {
+		if p.checkPurpose(consent, vendor, vendorID, tcf1constants.Purpose(i)) {
```

Comment:
Does `checkPurpose` cover the following requirement:
```
20) If the host company has opted not to enforce a purpose in the local Prebid Server,
that configuration should not be considered in the legal basis determination for stripping
User IDs in the above requirement. i.e. if Prebid Server config says to ignore enforcement
of Purpose 2 or 7, that configuration should not automatically cause User IDs to always
be passed. 
```

Reply:
The question of enforcing a purpose is checked outside of `checkPurpose`, so at this level we are essentially ignoring the enforcement configs at this point in the code. So yes, we are taking care of point 20.